test/e2e/preemption/preemption_test.go

```diff
@@ -572,3 +656,11 @@ func createSandbox1SleepPodCofigs(cnt, time int) []k8s.SleepPodConfig {
 	}
 	return sandbox1Configs
 }
+
+func createSandbox1SleepPodCofigsWithStaticNode(cnt, time int) []k8s.SleepPodConfig {
+	sandbox1Configs := make([]k8s.SleepPodConfig, 0, cnt)
+	for i := 0; i < cnt; i++ {
+		sandbox1Configs = append(sandbox1Configs, k8s.SleepPodConfig{Name: fmt.Sprintf("sleepjob%d", i+1), NS: devNew, Mem: sleepPodMemLimit2, Time: time, Optedout: k8s.Allow, Labels: map[string]string{"queue": "root.sandbox1"}, RequiredNode: nodeName})
```

Comment:
This part is very suspicious. You're using the "RequiredNode" setting here, which will trigger the "simple" (aka. RequiredNode) preemption logic inside Yunikorn. That is not the generic preemption logic that Craig worked on. 

Please check if the console log of Yunikorn contains this:
```
log.Log(log.SchedApplication).Info("Triggering preemption process for daemon set ask",
		zap.String("ds allocation key", ask.GetAllocationKey()))
```

If this is the case (which is VERY likely), changes are necessary.

A trivial solution is to enhance the `RequiredNode` field, it has to be a struct like:
```
type RequiredNode struct {
  Node      string
  DaemonSet bool
}

type SleepPodConfig struct {
...
	Mem          int64
	RequiredNode RequiredNode 
	Optedout     AllowPreemptOpted
}
```
If DaemonSet == false, then this code doesn't run: https://github.com/apache/yunikorn-k8shim/blob/28995cd7289256be3d3b8f820d438ddec36778b3/test/e2e/framework/helpers/k8s/pod_conf.go#L76-L77

So this line effectively becomes:
```
sandbox1Configs = append(sandbox1Configs,
 k8s.SleepPodConfig{
    Name: fmt.Sprintf("sleepjob%d", i+1),
    NS: devNew, Mem: sleepPodMemLimit2,
    Time: time,
    Optedout: k8s.Allow,
    Labels: map[string]string{"queue": "root.sandbox1"},
    RequiredNode: RequiredNode{Node: nodeName, DeamonSet: false}
    }
)
```

The existing tests inside `simple_preemptor_test.go` are affected.

Reply:
Have you looked at this? Again, the tests should address YUNIKORN-2068. RequiredNodePreemptor does NOT use the predicates.